cli/internal/terraform/terraform.go

```diff
@@ -560,11 +467,6 @@ func (c *Client) applyManualStateMigrations(ctx context.Context) error {
 		}
 	}
 	return nil
-	// expects to be run on initialized workspace
```

Comment:
since the restriction of running init first still remains, we should bake this requirement into the code signature (e.g. accept an init result) or at least keep a warning at the top?

Reply:
A migration in the current code may do anything, not limited to interacting with a Terraform file, so I am strongly against adding any kind of hard requirements to executing this function.
In the worst case, this should simply thrown an error.

I think the comment was explicitly for the manual state migration that was commented out here, maybe @malt3 can comment on this?